server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -211,27 +210,21 @@ public void recoverToTarget(ActionListener<RecoveryResponse> listener) {
                 cancellableThreads,
                 logger
             );
-            final Engine.HistorySource historySource;
-            if (softDeletesEnabled && (shard.useRetentionLeasesInPeerRecovery() || retentionLeaseRef.get() != null)) {
-                historySource = Engine.HistorySource.INDEX;
-            } else {
-                historySource = Engine.HistorySource.TRANSLOG;
-            }
-            final Closeable retentionLock = shard.acquireHistoryRetentionLock(historySource);
+            final Closeable retentionLock = shard.acquireHistoryRetentionLock();
             resources.add(retentionLock);
             final long startingSeqNo;
             final boolean isSequenceNumberBasedRecovery = request.startingSeqNo() != SequenceNumbers.UNASSIGNED_SEQ_NO
                 && isTargetSameHistory()
-                && shard.hasCompleteHistoryOperations("peer-recovery", historySource, request.startingSeqNo())
-                && (historySource == Engine.HistorySource.TRANSLOG
+                && shard.hasCompleteHistoryOperations("peer-recovery", request.startingSeqNo())
```

Comment:
Very difficult to follow this `if`s forest, may be a few intermediate `boolean`s  would help?

Reply:
I think I'd rather do that in a follow up because some of this is going to get refactored away anyway with coming segrep changes